beacon-chain/core/state/transition.go

```diff
@@ -245,9 +245,35 @@ func ProcessSlots(ctx context.Context, state *pb.BeaconState, slot uint64) (*pb.
 		traceutil.AnnotateError(span, err)
 		return nil, err
 	}
+	highestSlot := state.Slot
+	var root [32]byte
+	var err error
+
+	if featureconfig.Get().EnableSkipSlotsCache {
+		// Restart from cached value, if one exists.
+		root, err = ssz.HashTreeRoot(state)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not HashTreeRoot(state)")
+		}
+		cached, ok := skipSlotCache.Get(root)
+		if ok && cached.(*pb.BeaconState).Slot <= slot {
+			state = proto.Clone(cached.(*pb.BeaconState)).(*pb.BeaconState)
+			highestSlot = state.Slot
+			skipSlotCacheHit.Inc()
+		} else {
+			skipSlotCacheMiss.Inc()
+		}
+	}
+
 	for state.Slot < slot {
 		if ctx.Err() != nil {
 			traceutil.AnnotateError(span, ctx.Err())
+			if featureconfig.Get().EnableSkipSlotsCache {
```

Comment:
I dont think you need this duplicated block.   L302 to L307 will always cover this case

Reply:
what about exceeded deadlines, wouldn't it not be captured then ?

Reply:
Oh i see. That's right